apps/admin-ui/src/spa/recurring-reservations/application-rounds/[id]/review/Review.tsx

```diff
@@ -245,8 +249,12 @@ export function Review({
     applicationRound?.reservationUnits?.flatMap((x) => x)
   );
 
-  const ds = getUnitOptions(resUnits);
-  const unitPks = uniqBy(ds, (unit) => unit.pk).sort((a, b) =>
+  const { user } = useSession();
+
+  const ds = getUnitOptions(resUnits).filter((unit) =>
+    hasPermission(user)(UserPermissionChoice.CanViewApplications, unit.pk)
+  );
```

Comment:
Do we need functions that return functions here?

Comment why we need to do frontend filtering => so we get unitOptions that are filtered (just permission checking isn't enough here).